packages/reports/addon/components/common-actions/schedule.ts

```diff
@@ -134,88 +144,105 @@ export default class ScheduleActionComponent extends Component<Args> {
+
+    assert('Delivery Rules are defined', this.deliveryRules);
+    this.deliveryRules = [...this.deliveryRules, newRule];
+    return newRule;
   }
 
   /**
    * @action doSave
    */
   @action
   async doSave() {
-    if (this.isRuleValid) {
-      // Only add relationships to the new delivery rule if the fields are valid
-      setProperties(this.localDeliveryRule, {
-        deliveredItem: this.args.model,
-        owner: this.user.getUser(),
-      });
-
-      this.attemptedSave = false;
-      let savePromise = new RSVP.Promise((resolve, reject) => {
-        assert('The localDeliveryRule is defined', this.localDeliveryRule);
-        this.args.onSave(this.localDeliveryRule, { resolve, reject });
-      });
-
-      try {
-        await savePromise;
-        assert('The localDeliveryRule is defined', this.localDeliveryRule);
-        this.naviNotifications.add({
-          title: `${capitalize(this.localDeliveryRule.deliveryType)} delivery schedule successfully saved!`,
-          style: 'success',
-          timeout: 'short',
+    this.deliveryRules?.forEach(async (rule) => {
+      if (rule.validations.isValid) {
+        // Only add relationships to the new delivery rule if the fields are valid
+        this.attemptedSave = false;
+        let savePromise = new RSVP.Promise((resolve, reject) => {
+          assert('The Rule is defined', rule);
+          this.args.onSave(rule, { resolve, reject });
         });
-        this.closeModal();
-      } catch (errors) {
-        this.notification = getAllApiErrMsg(errors.errors[0], 'There was an error updating your delivery settings');
-      } finally {
+
+        try {
+          await savePromise;
+          assert('The localDeliveryRule is defined', rule);
+          this.naviNotifications.add({
+            title: `'${capitalize(rule.name)}'  schedule successfully saved!`,
+            style: 'success',
+            timeout: 'short',
+          });
+          this.closeModal();
+        } catch (errors) {
+          this.notification = getAllApiErrMsg(
+            errors.errors[0],
+            `There was an error updating your delivery settings for schedule '${capitalize(rule.name)}'`
+          );
+        } finally {
+          this.isSaving = false;
+          this.attemptedSave = true;
+        }
+      } else {
         this.isSaving = false;
         this.attemptedSave = true;
       }
-    } else {
-      this.isSaving = false;
-      this.attemptedSave = true;
-    }
+    });
   }
 
   /**
    * @action doDelete
    */
   @action
-  async doDelete() {
-    const deliveryRule = this.localDeliveryRule;
+  async doDelete(deliveryRule: DeliveryRuleModel) {
     assert('The localDeliveryRule is defined', deliveryRule);
+    const name = deliveryRule.name;
     const deletePromise = new RSVP.Promise((resolve, reject) => {
       this.args.onDelete(deliveryRule, { resolve, reject });
     });
 
+    this.deliveryRules = this.deliveryRules?.filter((rule) => rule !== deliveryRule);
```

Comment:
What if the delete fails? Is this being done too soon? Do we need this? Is  `deliveryRulesForUser` not getting recomputed after the delete?